docs/hugo/content/contributing/extension-points/pre-reconciliation-owner-checker.md

```diff
@@ -0,0 +1,368 @@
+    return next(ctx, owner, resourceResolver, armClient, log)
+}
+```
+
+## Check Results
+
+The extension returns one of two results:
+
+### Proceed
+```go
+return extensions.ProceedWithReconcile(), nil
+```
+- Owner state permits reconciliation
+- Reconciliation will continue (GET will be attempted)
+- Normal reconciliation flow proceeds
+
+### Block
+```go
+return extensions.BlockReconcile("parent is updating"), nil
+```
+- Owner state blocks reconciliation
+- No GET or other operations attempted on the resource
+- Resource requeued to try again later
+- Condition set with the blocking reason
+
+### Error
+```go
+return extensions.PreReconcileCheckResult{}, fmt.Errorf("check failed: %w", err)
+```
+- The check itself failed
+- Error condition set on resource
+- Reconciliation blocked until error resolved
+
+## Reconciliation Flow
+
+Understanding where this fits in the reconciliation process:
+
+1. **Resource needs reconciliation**: Controller picks up resource
+2. **Owner resolution**: Owner resource identified and fetched
+3. **PreReconciliationOwnerChecker invoked**: Extension checks owner state **← YOU ARE HERE**
+4. **Decision point**:
+   - If **blocked**: Stop here, requeue, set condition
+   - If **proceed**: Continue to next step
+5. **GET resource**: Fetch resource from Azure (only if not blocked)
+6. **PreReconciliationChecker**: Check resource state (if implemented)
+7. **ARM operations**: PUT/PATCH/DELETE as needed
+
+## Testing
+
+When testing `PreReconciliationOwnerChecker` extensions:
+
+1. **Test with nil owner**: Verify handling when owner is nil
+2. **Test with wrong owner type**: Verify graceful handling
+3. **Test blocking states**: Cover all states that should block
+4. **Test proceed states**: Verify reconciliation proceeds when appropriate
+5. **Test error handling**: Verify proper error returns
+
+
+## Performance Considerations
+
+This extension is more efficient than `PreReconciliationChecker` when:
+
+- Owner checks are sufficient to make the decision
+- Avoiding GET saves API quota
+- Owner state changes less frequently than resource state
+- Multiple child resources share the same owner (check once, benefit many times)
+
+However, it's less flexible because:
+
+- You cannot examine the resource's current state
+- You must make decisions based only on owner information
+- You cannot access resource-specific fields or status
+
+## Important Notes
+
+- **No resource access**: You do **not** receive the resource being reconciled
+- **Owner only**: Make decisions based solely on owner state
+- **Earlier in flow**: Runs before GET, unlike PreReconciliationChecker
+- **Nil owner**: Always handle the nil owner case gracefully
+- **Type assertions**: Verify owner is expected type before accessing fields
+- **Call next()**: Call the next checker in the chain when checks pass
+- **Clear reasons**: Provide helpful blocking messages for users
+- **Logging**: Log decisions to help debugging
+
+## Real-World Scenarios
```

Comment:
Do we actually need this real world scenarios section? Can we combine with patterns section above?